modular_skyrat/master_files/code/modules/events/spacevine.dm

```diff
@@ -0,0 +1,776 @@
+
+/datum/spacevine_mutation/nitrogen_producing/on_grow(obj/structure/spacevine/holder)
+	var/turf/holder_turf = get_turf(holder)
+	holder_turf.atmos_spawn_air("n2=100;TEMP=100")
+
+// SPACE VINES (Note that this code is very similar to Biomass code)
+/obj/structure/spacevine
+	name = "space vines"
+	desc = "An extremely expansionistic species of vine."
+	icon = 'icons/effects/spacevines.dmi'
+	icon_state = "Light1"
+	anchored = TRUE
+	density = FALSE
+	layer = SPACEVINE_LAYER
+	mouse_opacity = MOUSE_OPACITY_OPAQUE //Clicking anywhere on the turf is good enough
+	pass_flags = PASSTABLE | PASSGRILLE
+	max_integrity = 50
+	var/energy = 0
+	var/datum/spacevine_controller/master = null
+	var/list/mutations = list()
+	var/plantbgone_resist = FALSE
+
+/obj/structure/spacevine/Initialize(mapload)
+	. = ..()
+	add_atom_colour("#ffffff", FIXED_COLOUR_PRIORITY)
+	AddElement(/datum/element/atmos_sensitive, mapload)
+
+/obj/structure/spacevine/examine(mob/user)
+	. = ..()
+	var/text = "This one is a"
+	if(mutations.len)
+		for(var/A in mutations)
+			var/datum/spacevine_mutation/SM = A
+			text += " [SM.name]"
+	else
+		text += " normal"
+	text += " vine."
+	. += text
+
+/obj/structure/spacevine/Destroy()
+	for(var/datum/spacevine_mutation/SM in mutations)
+		SM.on_death(src)
+	if(master)
+		master.VineDestroyed(src)
+	mutations = list()
+	set_opacity(0)
+	if(has_buckled_mobs())
+		unbuckle_all_mobs(force=1)
+	return ..()
+
+/obj/structure/spacevine/proc/on_chem_effect(datum/reagent/R)
+	var/override = 0
+	for(var/datum/spacevine_mutation/SM in mutations)
+		override += SM.on_chem(src, R)
+	if(!override && istype(R, /datum/reagent/toxin/plantbgone) && !plantbgone_resist)
+		if(prob(50))
+			qdel(src)
+
+/obj/structure/spacevine/proc/eat(mob/eater)
+	var/override = 0
+	for(var/datum/spacevine_mutation/SM in mutations)
+		override += SM.on_eat(src, eater)
+	if(!override)
+		qdel(src)
+
+/obj/structure/spacevine/attacked_by(obj/item/I, mob/living/user)
+	var/damage_dealt = I.force
+	if(I.get_sharpness())
+		damage_dealt *= 4
+	if(I.damtype == BURN)
+		damage_dealt *= 4
+
+	for(var/datum/spacevine_mutation/SM in mutations)
+		damage_dealt = SM.on_hit(src, user, I, damage_dealt) //on_hit now takes override damage as arg and returns new value for other mutations to permutate further
+	take_damage(damage_dealt, I.damtype, MELEE, 1)
+
+/obj/structure/spacevine/play_attack_sound(damage_amount, damage_type = BRUTE, damage_flag = 0)
+	switch(damage_type)
+		if(BRUTE)
+			if(damage_amount)
+				playsound(src, 'sound/weapons/slash.ogg', 50, TRUE)
+			else
+				playsound(src, 'sound/weapons/tap.ogg', 50, TRUE)
+		if(BURN)
+			playsound(src.loc, 'sound/items/welder.ogg', 100, TRUE)
+
+/obj/structure/spacevine/Crossed(atom/movable/AM)
+	. = ..()
+	if(!isliving(AM))
+		return
+	for(var/datum/spacevine_mutation/SM in mutations)
+		SM.on_cross(src, AM)
+	if(istype(AM, /mob/living/simple_animal/hostile/venus_human_trap)) //skyrat change: vines heal flytraps 10% on cross
```

Comment:
This isn't needed, they heal when attacking living people. This just lets them fullheal then go back and fight some more, and makes them far more difficult to kill when they are already not an easy target.